memcache_proxy/dtog.go

```diff
@@ -33,6 +33,12 @@ import (
 	pb "google.golang.org/appengine/notreallyinternal/memcache"
 )
 
+const (
+	// Max App Engine Memcache value size (10^6 bytes).
+	// https://cloud.google.com/appengine/docs/python/memcache/
+	maxMemcacheValueSize = 1000000
```

Comment:
Interesting. I'm tempted to focus on the 4K bug here and tackling this validation separately, as it might require doing proper testing across all languages to make sure we got it right. Let me know your thoughts.